src/urllib3/multipart/decoder.py

```diff
@@ -0,0 +1,149 @@
+"""Logic for parsing and decomposing a multipart response body."""
+from __future__ import annotations
+
+import email.parser
+import typing
+
+from .. import _collections
+from .. import response as _response
+from .encoder import encode_with
+
+
+class ImproperBodyPartContentError(Exception):
+    pass
+
+
+class NonMultipartContentTypeError(Exception):
+    pass
+
+
+def _header_parser(headers: bytes, encoding: str) -> typing.Sequence[tuple[str, str]]:
+    string = headers.decode(encoding)
+    items = email.parser.HeaderParser().parsestr(string).items()
+    items = typing.cast(typing.List[typing.Tuple[str, str]], items)
```

Comment:
I'm not sure this cast is necessary – if I try to reproduce in a minimal example, mypy doesn't complain when this line is removed.